### ISSUE TITLE
Support readOnly, writeOnly (>= Draft 7)

### ISSUE THREAD

**Issue by ahx** [davishmcclurg/json_schemer#55]
The `readOnly` and `writeOnly` keywords (see [specification (Draft 7)](https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-10.3)) make validation behave differently in the context of a write or read operation. 

Unfortunately this does not seem to be covered by [JSON-Schema-Test-Suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite).

I found one (Java) [library that supports this](https://github.com/everit-org/json-schema#readonly-and-writeonly-context).

**Comment by davishmcclurg** [comment]
The specification doesn't seem to get into how these keywords relate to validating data, though I can see how specifying a validation context could be useful. I don't plan on working on this myself, but PRs are always welcome.

**Comment by ahx** [comment]
There is an [example how one could handle `readOnly: true`](https://github.com/davishmcclurg/json_schemer/blob/master/test/json_schemer_test.rb#L230-L258) via the `before_property_validation` option that was added in https://github.com/davishmcclurg/json_schemer/pull/77 in one of the tests.

**Comment by ahx** [comment]
If someone is interested in this as well: Actually using the added hooks to hack in `readOnly` behaviour did not work in my case, because is not called on sub-schemas used inside `allOf` etc. :| 
Another, probably better, approach could be a custom keyword. There already is some support for custom keywords, but I cannot use that to hack readOnly.

**Comment by davishmcclurg** [comment]
Released in 2.0.0. Use `access_mode` option (read/write):

```ruby
JSONSchemer.schema(schema).validate(data, :access_mode => 'read')
```
